core/test/base/half.cpp

```diff
@@ -0,0 +1,285 @@
+// SPDX-FileCopyrightText: 2017 - 2024 The Ginkgo authors
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <bitset>
+#include <cstring>
+#include <string>
+
+#include <gtest/gtest.h>
+
+#include <ginkgo/core/base/half.hpp>
+
+
+template <std::size_t N>
+struct floating_impl;
+
+template <>
+struct floating_impl<16> {
+    using type = gko::half;
+};
+
+template <>
+struct floating_impl<32> {
+    using type = float;
+};
+
+template <>
+struct floating_impl<64> {
+    using type = double;
+};
+
+template <std::size_t N>
+using floating = typename floating_impl<N>::type;
```

Comment:
This is duplicated from `extended_float.cpp`. Maybe move this into a `floating.hpp` or `bit_helpers.hpp`, together with the `create_from_bits` and `get_bits` below.